users/serializers.py

```diff
@@ -37,6 +46,18 @@ class LegalAddressSerializer(serializers.ModelSerializer):
     state_or_territory = serializers.CharField(max_length=255, allow_blank=True)
     postal_code = serializers.CharField(max_length=10, allow_blank=True)
 
+    def validate_first_name(self, value):
+        """Validates the first name of the user"""
+        if value and not USER_NAME_RE.match(value):
```

Comment:
nit: As value is a required field, we can restrict that in the serializer as well.

Reply:
I didn't make this change in light of a comment made on this serializer for backward compatibility https://github.com/mitodl/mitxpro/blob/051c325660016e592c7a7e9616ca777b5608d714/users/serializers.py#L27